lib/omise/OmiseSearch.php

```diff
@@ -0,0 +1,224 @@
+     *
+     * @param  string $scope  See supported scope at [Search API](https://www.omise.co/search-api) page.
+     * @param  string $publickey
+     * @param  string $secretkey
+     *
+     * @return  OmiseSearch  The created instance.
+     */
+    public static function scope($scope, $publickey = null, $secretkey = null)
+    {
+        return new OmiseSearch($scope, $publickey, $secretkey);
+    }
+
+    /**
+     * Create an instance of `OmiseSearch` with the given scope.
+     *
+     * This constructor is `protected` thus not intended to be used directly.
+     *
+     * @param  string $scope  See supported scope at [Search API](https://www.omise.co/search-api) page.
+     * @param  string $publickey
+     * @param  string $secretkey
+     *
+     * @return  OmiseSearch  The created instance.
+     */
+    protected function __construct($scope, $publickey, $secretkey)
+    {
+        parent::__construct($publickey, $secretkey);
+        $this->mergeAttributes('scope', $scope);
+    }
+
+    /**
+     * Update `query` parameter.
+     *
+     * @param  string $query  Searching text within the scope.
+     *
+     * @return  OmiseSearch  This instance.
+     */
+    public function query($query)
+    {
+        return $this->mergeAttributes('query', $query);
+    }
+
+    /**
+     * Update `filters` parameter.
+     *
+     * @param  string $filters  Searching text with specific key within the scope.
+     *
+     * @return  OmiseSearch  This instance.
+     */
+    public function filter(array $filters = array())
+    {
+        foreach ($filters as $k => $v) {
+            if (is_bool($v)) {
+                $filters[$k] = $v ? 'true' : 'false';
+            }
+        }
+        return $this->mergeAttributes('filters', $filters);
+    }
+
+    /**
+     * Update `page` parameter.
+     *
+     * @param  string $page  Specific number of searching page.
+     *
+     * @return  OmiseSearch  This instance.
+     */
+    public function page($page)
+    {
+        return $this->mergeAttributes('page', $page);
+    }
+
+    /**
+     * Update `order` parameter.
+     *
+     * @param  string $order  The order of the list returned.
+     *
+     * @see https://www.omise.co/search-api
+     *
+     * @return  OmiseSearch  This instance.
+     */
+    public function order($order)
+    {
+        return $this->mergeAttributes('order', $order);
+    }
+
+    /**
+     * Check whether this search instance is dirty or not.
+     *
+     * Dirty search instance needs to be retrieved before further usage. The
+     * dirty search instance automatically retrieves its data when client code
+     * tries to access its array element.
+     *
+     *     $search = OmiseCharge::query('demo'); // the instance is dirty
+     *     echo $search['object'];               // this will automatically retrive remote value
+     *
+     * @return  bool  true if the instance is dirty and needs to be retrieved
+     */
+    public function isDirty() {
```

Comment:
@iwat ✋ have a question, in any usage that gonna have chance to call this method?